drivers/fpga/fpga_ice40.c

```diff
@@ -330,6 +330,10 @@ static int fpga_ice40_load_spi(const struct device *dev, uint32_t *image_ptr, ui
 	struct fpga_ice40_data *data = dev->data;
 	uint8_t clock_buf[(UINT8_MAX + 1) / BITS_PER_BYTE];
 	const struct fpga_ice40_config *config = dev->config;
+	struct spi_dt_spec bus;
+
+	memcpy(&bus, &config->bus, sizeof(bus));
+	memset(&bus.config.cs, 0, sizeof(bus.config.cs));
```

Comment:
Generally, this change looks good. However, this line looks a bit suspicious because there is a fair bit of detail in `spi_cs_control` that is getting set to null including printers.

It would be better to only change the exact fields that are required, which would be  `gpio_dt_spec.dt_flags`.

I think I did try that at one point, but the particular platform I was using (esp32-c3) did not seem to respond as expected due to some driver issue.

@benediktibk - what platform are you using for testing this?

Reply:
I'm testing this on a STM32F746. For this platform (as far as I understood it on all STM32) the check looks like this:

    static inline bool spi_cs_is_gpio(const struct spi_config *config)
    {
        return config->cs.gpio.port != NULL;
    }

So for the STM32 the `dt_flags` are irrelevant, only the port is checked. Is there some sort of consensus about the proper way to indicate that no CS-pin is set?

Reply:
@tbursztyka @teburd - any ideas?

Reply:
That diagram posted was extra helpful.

Is this 
```
 * CS  """""""""""""""""""""""""\________...________/"""""""""""""""""""""""""
 *
 * CLK """\_/\/\/\/\/\/\/\/\_/"""\_/\/\/\.../\/\/\_/"""\_/\/\/\/\/\/\/\/\_/"""
 *
 *        <--Leading phase--->   <--Data transfer-->   <--Trailing phase-->
```

Maybe some explanation of what the `memset()` does (set the GPIO pointer to NULL and clears CPHA/CPOL AFAIU) would be enough?

Reply:
The full sequence would include some additional quirkiness before the leading clock phase, but this diagram of yours correctly depicts the necessary "SPI"-phase.

I will add a comment to the memset about the why, this is a good idea.

Reply:
> I will add a comment to the memset about the why, this is a good idea.

done

Reply:
@cfriedt would it be enough compromise to `memset(0)` all fields to make sure the GPIO pin is not used, or would it be best to put `cs.gpio.port = NULL` specifically?

Reply:
Another way could be a `pinctrl` to disconnect the GPIO pin.

Reply:
I honestly do not care about how we disable the usage of the CS pin, I'm only concerned about consistency. And there we seem to lack some at the moment, therefore I would really value some input from the SPI guys @teburd @tbursztyka.

Reply:
memset() as it is done or specifically calling bus.config.cs.gpio.port = NULL are both fine to me (the later is probably the fastest). As long as spi_cs_is_gpio returns false you are good to go.

Reply:
@cfriedt Acceptable result for you?